src/lib/y2partitioner/actions/controllers/blk_device.rb

```diff
@@ -0,0 +1,147 @@
+# encoding: utf-8
+
+# Copyright (c) [2018] SUSE LLC
+#
+# All Rights Reserved.
+#
+# This program is free software; you can redistribute it and/or modify it
+# under the terms of version 2 of the GNU General Public License as published
+# by the Free Software Foundation.
+#
+# This program is distributed in the hope that it will be useful, but WITHOUT
+# ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+# FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License for
+# more details.
+#
+# You should have received a copy of the GNU General Public License along
+# with this program; if not, contact SUSE LLC.
+#
+# To contact SUSE LLC about this file by physical or electronic mail, you may
+# find current contact information at www.suse.com.
+
+require "yast"
+require "y2storage"
+require "y2partitioner/device_graphs"
+
+module Y2Partitioner
+  module Actions
+    module Controllers
+      # This class offers helper methods to perform actions over a block device,
+      # for example, to check if its filesystem exists on disk, it is mounted, etc.
+      class BlkDevice
+        # Block device
+        #
+        # @return [Y2Storage::BlkDevice]
+        attr_reader :device
+
+        # Constructor
+        #
+        # @raise [TypeError] if device is not a block device.
+        #
+        # @param device [Y2Storage::BlkDevice] it has to be a block device,
+        #   (i.e., device#is?(:blk_device) #=> true).
+        def initialize(device)
+          raise(TypeError, "param device has to be a block device") unless device.is?(:blk_device)
```

Comment:
NP: but if you do type protection, then maybe check also against nil and potential other misusage like string, so maybe something like `if !device.respond_to?(:is?) || !device.is?(:blk_device)`